src/staking-router-modules/staking-router.service.ts

```diff
@@ -61,15 +63,16 @@ export class StakingRouterService {
    * Helper method for getting staking module list and execution layer meta
    * @returns Staking modules list and execution layer meta
    */
-  public async getStakingModulesAndMeta(): Promise<{
+  public async getStakingModulesAndMeta(stakingModuleAddresses?: string[]): Promise<{
     stakingModules: SrModuleEntity[];
     elBlockSnapshot: ELBlockSnapshot;
   }> {
     const { stakingModules, elBlockSnapshot } = await this.entityManager.transactional(
       async () => {
-        const stakingModules = await this.getStakingModules();
+        const stakingModules = await this.getStakingModules(stakingModuleAddresses);
 
-        if (stakingModules.length === 0) {
+        // If the target query involves retrieving module-specific data, we do not throw the 425 exception
```

Comment:
@Amuhar  needs your attention. What do you think of this decision?

Reply:
i see that it is not possible to make it better now, so it is okay for now

Reply:
i checked it again
If db is not ready we have null el meta , so here better now to put check above stakingModules check and maybe we even will not need to check stakignModules list length than
```
 const elBlockSnapshot = await this.getElBlockSnapshot();
 if (!elBlockSnapshot) {
   this.logger.warn("Meta is null, maybe data hasn't been written in db yet");
   throw httpExceptionTooEarlyResp();
 }

```

Reply:
fixed